### PR TITLE
Event Changes

### PR DESCRIPTION
**Pull request by ivorisoutdoors**
- Add Request to AfterSendEvent
- Add Request and Response to ApiExceptionEvent

### CODE REVIEW

File: src/Event/ApiExceptionEvent.php [hunk 1]
```diff
@@ -22,13 +24,30 @@ class ApiExceptionEvent extends Event
...
+     * @param Exception         $exception
+     * @param RequestInterface  $request
+     * @param ResponseInterface $response
      */
-    public function __construct(Exception $exception)
-    {
+    public function __construct(
+        Exception $exception,
+        RequestInterface $request = null,
+        ResponseInterface $response = null
```
Comment (natebrunette) [inline src/Event/ApiExceptionEvent.php]: Will there ever be a response if an exception is thrown?

File: src/Event/AfterSendEvent.php [hunk 2]
```diff
@@ -22,13 +23,20 @@ class AfterSendEvent extends Event
...
+    private $request;
+
+    /**
      * Constructor
      *
      * @param ResponseInterface $response
+     * @param RequestInterface  $request
      */
-    public function __construct(ResponseInterface $response = null)
+    public function __construct(ResponseInterface $response = null, RequestInterface $request = null)
```
Comment (tonynelson19) [inline src/Event/AfterSendEvent.php]: Would it be better to switch the order of these arguments? And would they ever be `null`?

``` php
public function __construct(RequestInterface $request, ResponseInterface $response)
```

File: src/Event/ApiExceptionEvent.php (hunk 1 above)
Comment (ivorisoutdoors) [inline src/Event/ApiExceptionEvent.php]: Nope. I'll fix

File: src/Event/AfterSendEvent.php (hunk 2 above)
Comment (ivorisoutdoors) [inline src/Event/AfterSendEvent.php]: Switching the order makes sense to me. Not sure about them being null. @natebrunette?

Comment (ivorisoutdoors) [inline src/Event/AfterSendEvent.php]: If async, the response could be null.

**Pull request merged**